src/documentation/literal.rs

```diff
@@ -0,0 +1,443 @@
+use crate::{Range, Span};
+use anyhow::{anyhow, Result};
+
+use regex::Regex;
+use std::convert::TryFrom;
+use std::fmt;
+
+/// A literal with meta info where the first and list whitespace may be found.
+#[derive(Clone)]
+pub struct TrimmedLiteral {
+    /// The span of rendered content, minus pre and post already applied.
+    span: Span,
+    /// the complete rendered string including post and pre.
+    rendered: String,
+    /// Literal prefx
+    pub pre: usize,
+    /// Literal postfix
+    pub post: usize,
+    /// Length of rendered **minus** `pre` and `post`.
+    /// If the literal is all empty, `pre` and `post` become `0`, and `len` covers the full length of `rendered`.
+    len: usize,
+}
+
+impl std::cmp::PartialEq for TrimmedLiteral {
+    fn eq(&self, other: &Self) -> bool {
+        if self.rendered != other.rendered {
+            return false;
+        }
+        if self.pre != other.pre {
+            return false;
+        }
+        if self.post != other.post {
+            return false;
+        }
+        if self.len() != other.len() {
+            return false;
+        }
+        if self.span != other.span {
+            return false;
+        }
+
+        true
+    }
+}
+
+impl std::cmp::Eq for TrimmedLiteral {}
+
+impl std::hash::Hash for TrimmedLiteral {
+    fn hash<H: std::hash::Hasher>(&self, hasher: &mut H) {
+        self.rendered.hash(hasher);
+        self.span.hash(hasher);
+        self.pre.hash(hasher);
+        self.post.hash(hasher);
+        self.len.hash(hasher);
+    }
+}
+
+impl TryFrom<proc_macro2::Literal> for TrimmedLiteral {
+    type Error = anyhow::Error;
+    fn try_from(literal: proc_macro2::Literal) -> Result<Self> {
+        let rendered = dbg!(literal.to_string());
+
+        lazy_static::lazy_static! {
+            static ref PREFIX_ERASER: Regex = Regex::new(r##"^((?:r#*)?")"##).unwrap();
+            static ref SUFFIX_ERASER: Regex = Regex::new(r##"("#*)$"##).unwrap();
+        };
+
+        let pre = if let Some(captures) = PREFIX_ERASER.captures(rendered.as_str()) {
+            if let Some(prefix) = captures.get(1) {
+                prefix.as_str().len()
+            } else {
+                return Err(anyhow!("Unknown prefix of literal"));
+            }
+        } else {
+            return Err(anyhow!("Missing prefix of literal: {}", rendered.as_str()));
+        };
+        let post = if let Some(captures) = SUFFIX_ERASER.captures(rendered.as_str()) {
+            // capture indices are 1 based, 0 is the full string
+            if let Some(suffix) = captures.get(captures.len() - 1) {
+                suffix.as_str().len()
+            } else {
+                return Err(anyhow!("Unknown suffix of literal"));
+            }
+        } else {
+            return Err(anyhow!("Missing suffix of literal: {}", rendered.as_str()));
+        };
+        let _scrap = |c: &'_ char| -> bool { c.is_whitespace() };
```

Comment:
Whitespaces are ignored now and not counted for `pre`?

Reply:
Well, `_scrap` is dead, should be removed...